core/src/main/java/com/exadel/aem/backpack/core/services/impl/PackageServiceImpl.java

```diff
@@ -1047,4 +1072,55 @@ private boolean isFolder(String resourceType) {
         return packageInfos;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public JcrPackageWrapper uploadPackage(final Session session,
+                                           final byte[] fileUploadBytesArray,
+                                           final boolean forceUpdate) {
+        JcrPackageWrapper jcrPackageWrapper = new JcrPackageWrapper();
+        File fileUpload = null;
+
+        if (session != null && fileUploadBytesArray != null) {
+            JcrPackageManager packageManager = getPackageManager(session);
+            try {
+                fileUpload = getFile(fileUploadBytesArray);
+                String nameHint = "uploaded_package_name";
+                final boolean isTempFile = true;
+                final boolean strict = true;
+
+                JcrPackage upload = packageManager.upload(fileUpload, isTempFile, forceUpdate, nameHint, strict);
+                jcrPackageWrapper.setJcrPackage(upload);
+
+                return jcrPackageWrapper;
+            } catch (Exception e) {
+                LOGGER.error("Cannot upload package: {}", e.getMessage(), e);
```

Comment:
Maybe it makes sense to put package name here instead of e.getMessage()

Reply:
In this case we will have more extended msg about error